tests/roles/barbican_adoption/tasks/main.yaml

```diff
@@ -0,0 +1,54 @@
+- name: deploy podified Barbican
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    spec:
+      barbican:
+        enabled: true
+        apiOverride:
+          route: {}
+        template:
+          databaseInstance: openstack
+          secret: osp-secret
+          barbicanAPI:
+            replicas: 1
+            override:
+              service:
+                internal:
+                  metadata:
+                    annotations:
+                      metallb.universe.tf/address-pool: internalapi
+                      metallb.universe.tf/allow-shared-ip: internalapi
+                      metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+                  spec:
+                    type: LoadBalancer
+          barbicanWorker:
+            replicas: 1
+          barbicanKeystoneListener:
+            replicas: 1
+    '
+
+- name: wait for Barbican to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=service=barbican -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+  register: barbican_running_result
+  until: barbican_running_result is success
+  retries: 180
+  delay: 2
+
+- name: check that Barbican is reachable and its endpoints are defined
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+
+    alias openstack="oc exec -t openstackclient -- openstack"
+
+    ${BASH_ALIASES[openstack]} endpoint list | grep key-manager
+    ${BASH_ALIASES[openstack]} secret list
+  register: barbican_responding_result
+  until: barbican_responding_result is success
+  retries: 60
+  delay: 2
```

Comment:
same question as above -- is there a way to create a secret before the migration - and then test for its existence here?  Also, we should probably do a test for creating a secret - that will test the operation of the worker.

It might be nice also to test the keystone-listener but that's going to involve a lot more setup.